rs/execution_environment/src/scheduler.rs

```diff
@@ -2061,6 +2077,10 @@ fn execute_canisters_on_thread(
                 &mut round_limits,
                 subnet_size,
             );
+            if instructions_used.unwrap_or(0.into()) > 0.into() {
```

Comment:
NIT: Maybe `instructions_used.map_or(false, |instructions| instructions > 0.into());` is more readable, I did not test it.

Reply:
Done, thank you!